### ISSUE TITLE
The selinux_enabled parameter should either accept a String or use the selinux.enabled fact

### ISSUE THREAD

**Issue by trevor-vaughan** [puppetlabs/puppetlabs-docker#212]
## What you expected to happen?

I need to be able to set the `selinux_enabled` parameter via Hiera but Hiera cannot pull `Boolean` values out of facts.

Previously, the module allowed a `String` value for the parameter but recently it was switched to a Boolean which simply won't work when setting data from Hiera.

## What happened?

```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Class[Docker]: parameter 'selinux_enabled' expects a value of type Undef or Boolean, got String
```

## How to reproduce it?

Set `docker::selinux_enabled` via Hiera

**Comment by ccaum** [comment]
Via @trevor-vaughan, an example Hiera config that reproduces the issue:
`docker::selinux_enabled: "%{facts.selinux.enabled}"`

**Comment by ccaum** [comment]
It looks like facts.selinux.enabled does in fact return a boolean, not a string. However, Hiera doesn't allow interpolation without string quotes? So essentially your Hiera config is casting the boolean to a string, @trevor-vaughan. 

What happens if you drop the quotes from the key's value? Does Hiera correctly interpolate?

**Comment by ccaum** [comment]
@trevor-vaughan I filed a Hiera bug (https://tickets.puppetlabs.com/browse/HI-599) to solve the underlying issue. I'm going to close this since it's a bug with Hiera.

**Comment by ccaum** [comment]
Thanks, @trevor-vaughan. When you say the module previously allowed a string value, are you referring to the garethr/docker module? If so, we didn't necessarily intend for all code to be perfectly portable to this module.

Is it acceptable  wrap the code in a profile and use the any2bool function to convert a `true` string a Boolean? https://forge.puppet.com/puppetlabs/stdlib#any2bool

**Comment by trevor-vaughan** [comment]
@ccaum I know that I can use a profile to do this. If you would like to update your documentation with an example profile that would work around this issue, that would technically suffice just make for a less than ideal user experience. I would suggest though, that the `selinux_enabled` Boolean should be handled by this module from the native Puppet fact since it *is* a native Puppet fact and needs to be handled carefully based on which version of Docker that you are installing. As a user, this is abstraction that I would expect the Docker module to do for me since it abstracts the rest of the interface.

**Comment by trevor-vaughan** [comment]
@ccaum Checking the code, 1.0.5 has `Optional[String]` and 1.1.0 has `Optional[Boolean]` https://github.com/puppetlabs/puppetlabs-docker/blob/1.0.5/manifests/init.pp#L399.

This is a breaking change since, previously you could set this value via Hiera without issue from our experience, and *if* you're following Semantic Versioning, would necessitate a 2.0.0 bump.

That said, it looks like, overall, it's a trivial fix since, as you pointed out, you just need to update the code to use `any2bool`.

**Comment by ccaum** [comment]
Hey @trevor-vaughan, we had a long discussion and investigation into this. The short of it is that the 1.0.5 release was a mistake that we should have dealt with when we realized our validation mistake. I definitely hear you about bumping the major version and I promise we'll be sensitive about interface changes going forward and we'll bump the major the version if a backwards breaking change to the  module's interface is made.

I recognize the need to get that Hiera limitation referenced above fixed. In the meantime, I think the best path forward is a profile as we discussed.
